### PR TITLE
Add article series

### LINKED ISSUE #215: Add article series
- Have a top-of-article admonition to link at the top of each post leading to the previous part(s) with something like "missed the previous articles?"
- Start off series with an intro article and end with outro article (for long series)
- Put links at the bottom of each post leading to the next part, so readers can easily go through the series. (Obviously, you’ll need to go back and add these in once you’ve published the later parts.)
- Add links to the body of the post whenever you mention a topic you’ve already written about in the series. (You can also go back to earlier posts and link them to later posts in the series.) This saves you having to repeat yourself a lot, and makes it easy for readers to find the information they need at any given point.
- Start article with quick recap and end with coming soon preview
- CTA to subscribe to stay up to date with new releases
Image

### PR DESCRIPTION
**Pull request by andreilgeorgescu**
Resolves #215

- [x] Create series components
  - [x] Top-of-article
  - [x] Bottom-of-article
- [x] Add series components to `ArticleLayout`
- [x] Add series to article schema
- [x] Create series content collections (@daniel-ciocirlan) (I added the obvious ones)
- [x] Newsletter component at the bottom

### CODE REVIEW

**Comment by andreilgeorgescu** [comment]
- [ ] Change text to "This article is apart of a series"
- [ ] Gradient should be strong at top
- [ ] Gradient should merge into different color
- [ ] Add "This article" to series article indicator
- [ ] Make dropdown arrow more prominent

**Comment by andreilgeorgescu** [comment]
@daniel-ciocirlan I've added everything we discussed.

I ended up removing the arrow because it's fairly difficult to style, and frankly, I think it looks better without it. What do you think?

Do you still want a different gradient color?

You can add any remaining series content collections.

The newsletter section needs to be hooked up to your newsletter service. The Mailchimp script actually seems to cause lighthouse issues due to blocking (despite being deferred for some reason). Since we wanted to migrate you from MailChimp to MailerLite anyway, this seems like a good time as any. You can create a free account on [MailerLite](mailerlite.com) and then I can set up a more efficient signup.

**Comment by daniel-ciocirlan** [comment]
Looks good, with a couple of suggestions:
- spell "apart"
- can remove the gradient and just keep the --color-secondary as the backdrop
- instead of having a proper arrow image and bend over backwards with styling, you can use an ASCII character like ▶ or ▼ directly in the text

**Comment by wkenned1** [comment]
- [x] Remove gradient
- [x] Add arrow characters
- [x] "apart" -> "part"
- [x] Move to React component for click handling

**Comment by daniel-ciocirlan** [comment]
Here are the article series I could find:

Functional Error Handling in Kotlin
- functional-error-handling-in-kotlin-part-1-absent-values
- functional-error-handling-in-kotlin-part-2-result-and-either
- functional-error-handling-in-kotlin-part-3-the-raise-dsl

Authentication with Scala and Http4s
- authentication-with-scala-and-http4s
- authentication-with-scala-and-http4s-oauth
- authentication-with-scala-and-http4s-oauth-two-factor-authentication

Websockets in Scala (should we re-spell the URL? websockers -> websockets)
- websockers-in-scala-part-1-http4s
- websockers-in-scala-part-2-integrating-redis-and-postgresql

Kafka 101
- kafka-101-clients
- kafka-101-streams

Kotlin 101
- kotlin-101-coroutines
- kotlin-101-flows
- kotlin-101-type-classes
- kotlin-101-context-receivers

Type-Level Programming in Scala 3
- type-level-programming-in-scala-3-part-1-comparing-types
- type-level-programming-in-scala-3-part-2-a-quicksort-on-types

Type-Level Programming in Scala 2
- type-level-programming-in-scala-part-1-numbers-and-comparisons
- type-level-programming-in-scala-part-2-adding-numbers-as-types
- type-level-programming-in-scala-part-3-sorting-lists

**Review by daniel-ciocirlan — APPROVED** [review]
Looks good! A small rephrasing suggestion, to make the message warmer and friendlier:
- "Want more content like this? Sign up for our newsletter." -> "Like what you see? Keep learning by subscribing to the newsletter!"
- "We care about your data. Read our privacy policy" -> "This site is built on trust, and your data is safe. Check out the privacy policy"

File: src/pages/articles/_components/NewsletterSection.astro [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+        <button
+          type="submit"
+          class="flex-none rounded-md bg-cta px-3.5 py-2.5 text-sm font-semibold text-white shadow-sm hover:bg-accent-1 focus-visible:outline focus-visible:outline-2 focus-visible:outline-offset-2 focus-visible:outline-indigo-600"
+          >Subscribe</button
+        >
+      </div>
+      <p class="mt-4 text-sm/6 text-content-1">
+        This site is built on trust, and your data is safe. Check out the <a
+          href="/legal/privacy"
+          class="font-semibold">privacy&nbsp;policy</a
```
Comment (andreilgeorgescu) [inline src/pages/articles/_components/NewsletterSection.astro:31]:
```suggestion
          href="/legal/privacy"
          target="_blank"
          rel="noreferrer"
          class="font-semibold">privacy&nbsp;policy</a
```

**Pull request merged**